accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/aem/AemCqActionsSupportImpl.java

```diff
@@ -8,60 +8,253 @@
  */
 package biz.netcentric.cq.tools.actool.aem;
 
+import java.security.AccessControlException;
 import java.security.Principal;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 
 import javax.jcr.Node;
 import javax.jcr.RepositoryException;
 import javax.jcr.Session;
+import javax.jcr.Value;
+import javax.jcr.nodetype.NodeDefinition;
+import javax.jcr.nodetype.NodeType;
+import javax.jcr.security.AccessControlManager;
 import javax.jcr.security.Privilege;
 
+import org.apache.jackrabbit.JcrConstants;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlList;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlManager;
 import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import com.day.cq.security.util.CqActions;
+import biz.netcentric.cq.tools.actool.configmodel.AceBean;
+import biz.netcentric.cq.tools.actool.helper.AccessControlUtils;
 
 @Component(service = AemCqActionsSupport.class)
 public class AemCqActionsSupportImpl implements AemCqActionsSupport {
+    private static final Logger LOG = LoggerFactory.getLogger(AemCqActionsSupportImpl.class);
 
-    public static class CqActionsWrapper implements AemCqActions {
-        
-        private final CqActions cqActions;
+    public enum CqActions {
+        read, modify, create, delete, acl_read, acl_edit, replicate;
+    }
 
-        public CqActionsWrapper(com.day.cq.security.util.CqActions cqActions) {
-            this.cqActions = cqActions;
-        }
+    public static final String REPLICATE_PRIVILEGE = "{http://www.day.com/crx/1.0}replicate";
```

Comment:
Let us reuse this constant: https://www.adobe.io/experience-manager/reference-materials/6-5/javadoc/com/day/cq/replication/Replicator.html#REPLICATE_PRIVILEGE